pandas/_libs/tslibs/nattype.pyx

```diff
@@ -1210,6 +1211,22 @@ default 'raise'
     def tzinfo(self) -> None:
         return None
 
+    def as_unit(self, str unit, bint round_ok=True) -> "NaTType":
+        """
+        Convert the underlying int64 representaton to the given unit.
+
+        Parameters
+        ----------
+        unit : {"ns", "us", "ms", "s"}
+        round_ok : bool, default True
+            If False and the conversion requires rounding, raise.
+
+        Returns
```

Comment:
Could you add an `Examples` section too? (Hoping to have all public APIs have examples in the near future)

Reply:
that would make sense, but first i think we should determine if we want unit to somehow show up in the repr

Reply:
Hmm, frequency didn't end up in the repr before right? (just as a comparison note)

Reply:
In 1.5.x `freq` is in the repr when it is non-None

Reply:
Might be helpful then to have the unit resolution show up in the repr then (guess it would be an API change)

Reply:
yah that's on my todo list, just not a priority ATM